compiler/src/iree/compiler/Codegen/Common/VectorReductionToGPU.cpp

```diff
@@ -80,7 +81,8 @@ static Value groupReduction(Location loc, OpBuilder &builder, Value input,
     builder.create<gpu::BarrierOp>(loc);
     VectorType vecType = VectorType::get(numWarp, input.getType());
     Value zero = builder.create<arith::ConstantIndexOp>(loc, 0);
-    Value vec = builder.create<vector::LoadOp>(loc, vecType, alloc, zero);
+    Value vec =
+        builder.create<vector::TransferReadOp>(loc, vecType, alloc, zero);
```

Comment:
Why do we need to change this to transfer_read?

Reply:
This is needed to make the `SPIRVVectorizeLoadStore` pass work. It expects transfer ops to be able to change the memref of scalars into memref of vectors.

Reply:
Any chance we can extend `SPIRVVectorizeLoadStore` instead or is it too complex? (vector.load should be mainly a subset of transfer_read)

Reply:
I thought about extending to support `vector.read` op. Yeah it's much more complicated than doing this and let lowering to handle the generation of `vector.read` ops. Right now for `vector.transfer_read` we have something like https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorizeLoadStore.cpp#L196-L255, it's a 50 liner even with utility/base classes. `vector.read` should be simpler than that I guess but still quite some additional code to maintain. By introducing support for `vector.read` we'd also need to think about hybrid transfer ops and normal ops together, as adjusting the memref considers all users. There are also things like different bitwidth to consider..

Any reason we cannot generate transfer ops here?

Reply:
This is probably fine here. My concern is that not having support for some of the basic `vector` operations makes common transformations between SPIR-V and other backends a bit fragile in my opinion. If we start sharing more passes I think this may become a problem in the future.

Reply:
I see you points. I'll keep it as is for this time; next time if similar things arise I'll see how to make it work there. :)